File: static/js/doctor.js
// HeyMira - Doctor Dashboard

let sentimentTimelineChart = null;
let riskDistributionChart = null;
let messageSentimentChart = null;
let currentPatients = [];

document.addEventListener('DOMContentLoaded', async () => {
    await loadDoctorInfo();
    await loadPatients();
    await loadAlerts();
    await loadAvailablePatients();

    // Poll for new alerts every 60 seconds
    setInterval(loadAlerts, 60000);
});

// ========== Tab System ==========

function switchTab(tab) {
    document.querySelectorAll('.tab-btn').forEach(btn => {
        btn.classList.toggle('active', btn.dataset.tab === tab);
    });
    document.querySelectorAll('.tab-content').forEach(content => {
        content.classList.toggle('active', content.id === `tab-${tab}`);
    });

    if (tab === 'find') loadAvailablePatients();
    if (tab === 'analytics') populateAnalyticsDropdown();
}

// ========== Doctor Info ==========

async function loadDoctorInfo() {
    try {
        const data = await apiCall('/api/auth/me');
        document.getElementById('doc-name').textContent = `Dr. ${data.user.username}`;
        if (data.user.profile_image) {
            document.getElementById('doc-avatar').innerHTML = `<img src="${data.user.profile_image}" style="width:100%;height:100%;border-radius:50%;object-fit:cover;">`;
        } else {
            document.getElementById('doc-avatar').textContent = data.user.username[0].toUpperCase();
        }
        if (data.user.theme) setTheme(data.user.theme);
    } catch (error) {
        window.location.href = '/login';
    }
}

// ========== My Patients ==========

async function loadPatients() {
    try {
        const data = await apiCall('/api/doctor/patients');
        currentPatients = data.patients;
        const list = document.getElementById('patient-list');

        if (!data.patients.length) {
            list.innerHTML = '<div class="empty-state"><div class="empty-icon">👥</div><p>No patients connected yet. Go to "Find Patients" to send connection requests.</p></div>';
            return;
        }

        list.innerHTML = data.patients.map(p => {
            const risk = p.latest_report ? p.latest_report.risk_level : 'low';
            const sentiment = p.latest_report ? p.latest_report.sentiment_score : 0;
            return `
                <div class="patient-card" onclick="viewPatient(${p.user.id}, '${escapeHtml(p.user.username)}')">
                    <div class="patient-card-left">
                        <div class="patient-avatar">${p.user.username[0].toUpperCase()}</div>
                        <div class="patient-info">
                            <h4>${escapeHtml(p.user.username)}</h4>
                            <p>${p.total_conversations} sessions · ${getMoodEmoji(sentiment)} ${sentiment.toFixed(1)}</p>
                        </div>
                    </div>
                    <div class="patient-card-right">
                        ${p.unread_alerts > 0 ? `<span class="badge badge-critical">⚠ ${p.unread_alerts}</span>` : ''}
                        <span class="badge ${getRiskBadge(risk)}">${risk}</span>
                    </div>
                </div>
            `;
        }).join('');
    } catch (error) {
        showToast('Failed to load patients', 'error');
    }
}

// ========== Find & Connect Patients ==========

async function loadAvailablePatients() {
    try {
        const data = await apiCall('/api/doctor/available-patients');
        const list = document.getElementById('available-patients-list');

        if (!data.patients.length) {
            list.innerHTML = '<div class="empty-state"><div class="empty-icon">👥</div><p>No patients registered yet</p></div>';
            return;
        }

        list.innerHTML = data.patients.map(p => {
            let statusHtml = '';
            if (p.is_connected) {
                statusHtml = '<span class="badge badge-low">✅ Connected</span>';
            } else if (p.connection_status === 'pending') {
                statusHtml = '<span class="badge badge-moderate">⏳ Pending</span>';
            } else {
                statusHtml = `<button class="btn btn-primary btn-sm" onclick="event.stopPropagation(); sendRequest(${p.user.id})">Send Request</button>`;
            }

            return `
                <div class="patient-card" style="cursor:default;">
                    <div class="patient-card-left">
                        <div class="patient-avatar">${p.user.username[0].toUpperCase()}</div>
                        <div class="patient-info">
                            <h4>${escapeHtml(p.user.username)}</h4>
                            <p>${p.user.email} · Joined ${formatDate(p.user.created_at)}</p>
                        </div>
                    </div>
                    <div class="patient-card-right">
                        ${statusHtml}
                    </div>
                </div>
            `;
        }).join('');
    } catch (error) {
        showToast('Failed to load available patients', 'error');
    }
}

async function sendRequest(patientId) {
    try {
        await apiCall(`/api/doctor/connect/${patientId}`, {
            method: 'POST',
            body: JSON.stringify({ message: 'I would like to help support your mental wellness journey.' })
        });
        showToast('Connection request sent!', 'success');
        await loadAvailablePatients();
    } catch (error) {
        showToast(error.message, 'error');
    }
}

// ========== Alerts ==========

async function loadAlerts() {
    try {
        const data = await apiCall('/api/doctor/alerts');
        const list = document.getElementById('alert-list');
        const countEl = document.getElementById('alert-count');

        const unread = data.alerts.filter(a => !a.is_read);
        if (unread.length > 0) {
            countEl.textContent = unread.length;
            countEl.classList.add('visible');
        } else {
            countEl.classList.remove('visible');
        }

        if (!data.alerts.length) {
            list.innerHTML = '<div class="empty-state"><div class="empty-icon">✅</div><p>No alerts — all patients are doing well</p></div>';
            return;
        }

        list.innerHTML = data.alerts.slice(0, 20).map(a => `
            <div class="alert-item ${a.is_read ? '' : 'unread'}">
                <div class="alert-header">
                    <span class="alert-type">🚨 ${a.alert_type}</span>
                    <span class="alert-time">${formatDate(a.created_at)}</span>
                </div>
                <div class="alert-body">
                    <strong>${a.patient_name}</strong>: ${escapeHtml(a.message)}
                </div>
                ${!a.is_read ? `<button class="btn btn-sm btn-secondary" style="margin-top:8px;" onclick="markAlertRead(${a.id})">Mark as Read</button>` : ''}
            </div>
        `).join('');
    } catch (e) { }
}

async function markAlertRead(alertId) {
    try {
        await apiCall(`/api/doctor/alerts/${alertId}/read`, { method: 'POST' });
        await loadAlerts();
        showToast('Alert marked as read', 'success');
    } catch (error) {
        showToast(error.message, 'error');
    }
}

// ========== Patient Detail Modal ==========

async function viewPatient(patientId, name) {
    const modal = document.getElementById('patient-modal');
    document.getElementById('modal-patient-name').textContent = `${name}'s Details`;
    modal.classList.add('active');

    // Hide messages section initially
    document.getElementById('modal-messages-section').style.display = 'none';

    // Load conversations
    try {
        const convData = await apiCall(`/api/doctor/patient/${patientId}/conversations`);
        const convList = document.getElementById('modal-conversations');

        if (!convData.conversations.length) {
            convList.innerHTML = '<div class="empty-state"><p>No conversations yet</p></div>';
        } else {
            convList.innerHTML = convData.conversations.map(c => `
                <div class="patient-card" style="margin-bottom:8px;cursor:pointer;" onclick="viewConversationMessages(${patientId}, ${c.id}, '${escapeHtml(c.title)}')">
                    <div class="patient-card-left">
                        <div style="font-size:1.2rem;">💬</div>
                        <div class="patient-info">
                            <h4 style="font-size:0.9rem;">${escapeHtml(c.title)}</h4>
                            <p>${formatDate(c.started_at)} · ${c.message_count} messages</p>
                        </div>
                    </div>
                    <div class="patient-card-right">
                        <span class="badge ${getRiskBadge(c.risk_level)}">${c.risk_level}</span>
                        <span style="font-size:0.8rem;color:var(--text-muted);">${getMoodEmoji(c.sentiment_score)} ${c.sentiment_score.toFixed(1)}</span>
                    </div>
                </div>
            `).join('');
        }
    } catch (e) {
        document.getElementById('modal-conversations').innerHTML = '<p style="color:var(--error);">Failed to load conversations</p>';
    }

    // Load reports
    try {
        const repData = await apiCall(`/api/doctor/patient/${patientId}/reports`);
        const reports = document.getElementById('modal-reports');

        if (!repData.reports.length) {
            reports.innerHTML = '<div class="empty-state"><p>No reports available yet</p></div>';
        } else {
            reports.innerHTML = repData.reports.map(r => `
                <div class="report-item ${r.risk_level === 'high' || r.risk_level === 'critical' ? 'high' : r.risk_level === 'moderate' ? 'moderate' : ''}">
                    <div class="report-date">${formatDate(r.created_at)}</div>
                    <div class="report-summary">${escapeHtml(r.ai_summary)}</div>
                    <div class="report-meta">
                        <span class="badge ${getRiskBadge(r.risk_level)}">${r.risk_level}</span>
                        <span class="mood-indicator">${getMoodEmoji(r.sentiment_score)} ${r.sentiment_score.toFixed(2)}</span>
                        <span style="font-size:0.8rem;color:var(--text-muted);">Trend: ${r.emotional_trend}</span>
                    </div>
                </div>
            `).join('');
        }
    } catch (e) {
        document.getElementById('modal-reports').innerHTML = '<p style="color:var(--error);">Failed to load reports</p>';
    }
}

async function viewConversationMessages(patientId, convId, title) {
    const section = document.getElementById('modal-messages-section');
    const container = document.getElementById('modal-messages');
    section.style.display = 'block';

    container.innerHTML = '<p>Loading messages...</p>';

    try {
        const data = await apiCall(`/api/doctor/patient/${patientId}/conversation/${convId}/messages`);

        if (!data.messages.length) {
            container.innerHTML = '<div class="empty-state"><p>No messages in this conversation</p></div>';
            return;
        }

        container.innerHTML = data.messages.map(m => `
            <div class="msg-bubble ${m.role === 'user' ? 'user-msg' : 'ai-msg'}">
                <div>${escapeHtml(m.content)}</div>
                <div class="msg-meta">
                    ${m.is_voice ? '🎤 ' : ''}${formatTime(m.timestamp)}
                    ${m.role === 'user' && m.sentiment_score ? ` · ${getMoodEmoji(m.sentiment_score)} ${m.sentiment_score.toFixed(1)}` : ''}
                </div>
            </div>
        `).join('');

        section.scrollIntoView({ behavior: 'smooth' });
    } catch (error) {
        container.innerHTML = '<p style="color:var(--error);">Failed to load messages</p>';
    }
}

function closeModal() {
    document.getElementById('patient-modal').classList.remove('active');
}

document.addEventListener('click', (e) => {
    if (e.target.id === 'patient-modal') closeModal();
});

// ========== Analytics ==========

function populateAnalyticsDropdown() {
    const select = document.getElementById('analytics-patient-select');
    const currentVal = select.value;
    select.innerHTML = '<option value="">Select a patient...</option>';
    currentPatients.forEach(p => {
        const opt = document.createElement('option');
        opt.value = p.user.id;
        opt.textContent = p.user.username;
        select.appendChild(opt);
    });
    if (currentVal) select.value = currentVal;
}

async function loadPatientAnalytics(patientId) {
    if (!patientId) {
        document.getElementById('analytics-content').style.display = 'none';
        document.getElementById('analytics-empty').style.display = 'block';
        return;
    }

    try {
        const data = await apiCall(`/api/doctor/patient/${patientId}/analytics`);

        document.getElementById('analytics-empty').style.display = 'none';
        document.getElementById('analytics-content').style.display = 'block';

        // Render summary cards
        const summary = data.summary;
        const riskColor = summary.current_risk === 'critical' ? 'var(--error)' :
            summary.current_risk === 'high' ? '#f59e0b' :
                summary.current_risk === 'moderate' ? '#eab308' : 'var(--success)';

        document.getElementById('analytics-summary').innerHTML = `
            <div class="summary-card">
                <div class="card-value">${summary.total_sessions}</div>
                <div class="card-label">Sessions</div>
            </div>
            <div class="summary-card">
                <div class="card-value">${summary.total_messages}</div>
                <div class="card-label">Messages</div>
            </div>
            <div class="summary-card">
                <div class="card-value" style="color:${riskColor};">${getMoodEmoji(summary.average_sentiment)} ${summary.average_sentiment.toFixed(2)}</div>
                <div class="card-label">Avg Sentiment</div>
            </div>
            <div class="summary-card">
                <div class="card-value" style="color:var(--error);">${summary.crisis_alerts}</div>
                <div class="card-label">Crisis Alerts</div>
            </div>
            <div class="summary-card">
                <div class="card-value" style="color:${riskColor};">${summary.current_risk.toUpperCase()}</div>
                <div class="card-label">Current Risk</div>
            </div>
            <div class="summary-card">
                <div class="card-value">${summary.current_trend === 'improving' ? '📈' : summary.current_trend === 'declining' ? '📉' : '➡️'}</div>
                <div class="card-label">${summary.current_trend}</div>
            </div>
        `;

        // Build charts
        buildSentimentTimelineChart(data.sentiment_timeline);
        buildRiskDistributionChart(data.risk_distribution);
        buildMessageSentimentChart(data.message_sentiments);

    } catch (error) {
        showToast('Failed to load analytics: ' + error.message, 'error');
    }
}

function buildSentimentTimelineChart(timeline) {
    const ctx = document.getElementById('sentiment-timeline-chart');
    if (sentimentTimelineChart) sentimentTimelineChart.destroy();

    if (!timeline.length) {
        sentimentTimelineChart = null;
        return;
    }

    const labels = timeline.map(t => {
        const d = new Date(t.date + (t.date.endsWith('Z') ? '' : 'Z'));
        return d.toLocaleDateString('en-IN', { day: 'numeric', month: 'short' });
    });
    const scores = timeline.map(t => t.score);

    sentimentTimelineChart = new Chart(ctx, {
        type: 'line',
        data: {
            labels,
            datasets: [{
                label: 'Sentiment Score',
                data: scores,
                borderColor: '#a855f7',
                backgroundColor: 'rgba(168, 85, 247, 0.1)',
                fill: true,
                tension: 0.4,
                pointBackgroundColor: scores.map(s => s < -0.3 ? '#ef4444' : s < 0 ? '#f59e0b' : '#10b981'),
                pointBorderColor: 'transparent',
                pointRadius: 6,
                borderWidth: 3
            }]
        },
        options: {
            responsive: true,
            maintainAspectRatio: false,
            scales: {
                y: {
                    min: -1, max: 1,
                    ticks: { color: '#6b6d80', font: { family: 'Inter' } },
                    grid: { color: 'rgba(255,255,255,0.05)' }
                },
                x: {
                    ticks: { color: '#a7a9be', font: { family: 'Inter' } },
                    grid: { display: false }
                }
            },
            plugins: {
                legend: { display: false },
                tooltip: {
                    callbacks: {
                        afterLabel: (context) => {
                            const t = timeline[context.dataIndex];
                            return `Risk: ${t.risk_level} | Trend: ${t.trend}`;
                        }
                    }
                }
            }
        }
    });
}

function buildRiskDistributionChart(distribution) {
    const ctx = document.getElementById('risk-distribution-chart');
    if (riskDistributionChart) riskDistributionChart.destroy();

    const labels = ['Low', 'Moderate', 'High', 'Critical'];
    const values = [distribution.low || 0, distribution.moderate || 0, distribution.high || 0, distribution.critical || 0];
    const colors = ['#10b981', '#eab308', '#f59e0b', '#ef4444'];

    riskDistributionChart = new Chart(ctx, {
        type: 'doughnut',
        data: {
            labels,
            datasets: [{
                data: values,
                backgroundColor: colors,
                borderWidth: 0,
                borderRadius: 4
            }]
        },
        options: {
            responsive: true,
            maintainAspectRatio: false,
            cutout: '65%',
            plugins: {
                legend: {
                    position: 'bottom',
                    labels: { color: '#a7a9be', font: { family: 'Inter', size: 11 }, padding: 12 }
                }
            }
        }
    });
}

function buildMessageSentimentChart(messages) {
    const ctx = document.getElementById('message-sentiment-chart');
    if (messageSentimentChart) messageSentimentChart.destroy();

    if (!messages.length) {
        messageSentimentChart = null;
        return;
    }

    const labels = messages.map((m, i) => `#${i + 1}`);
    const scores = messages.map(m => m.score);

    messageSentimentChart = new Chart(ctx, {
        type: 'bar',
        data: {
            labels,
            datasets: [{
                label: 'Sentiment',
                data: scores,
                backgroundColor: scores.map(s => {
                    if (s >= 0.3) return 'rgba(16, 185, 129, 0.7)';
                    if (s >= -0.2) return 'rgba(234, 179, 8, 0.7)';
                    return 'rgba(239, 68, 68, 0.7)';
                }),
                borderRadius: 4,
                barThickness: 8
            }]
        },
        options: {
            responsive: true,
            maintainAspectRatio: false,
            scales: {
                y: {
                    min: -1, max: 1,
                    ticks: { color: '#6b6d80', font: { size: 10 } },
                    grid: { color: 'rgba(255,255,255,0.03)' }
                },
                x: {
                    ticks: { color: '#6b6d80', font: { size: 9 }, maxRotation: 0 },
                    grid: { display: false }
                }
            },
            plugins: { legend: { display: false } }
        }
    });
}

function toggleAlerts() {
    switchTab('alerts');
}

function escapeHtml(text) {
    const div = document.createElement('div');
    div.textContent = text;
    return div.innerHTML;
}

File: routes/doctor.py
from flask import Blueprint, jsonify, request, current_app
from flask_login import login_required, current_user
from models.models import db, User, Alert, Report, Conversation, Message, ConnectionRequest
from datetime import datetime

doctor_bp = Blueprint('doctor', __name__)


def doctor_required(f):
    """Decorator to ensure only doctors can access."""
    from functools import wraps
    @wraps(f)
    def decorated(*args, **kwargs):
        if not current_user.is_authenticated or current_user.role != 'doctor':
            return jsonify({'error': 'Doctor access required'}), 403
        return f(*args, **kwargs)
    return decorated


# ========== Connection Requests ==========

@doctor_bp.route('/api/doctor/available-patients', methods=['GET'])
@login_required
@doctor_required
def get_available_patients():
    """Get all patients (for sending connection requests)."""
    all_patients = User.query.filter_by(role='patient').all()
    result = []
    for p in all_patients:
        # Check if already connected or request pending
        existing = ConnectionRequest.query.filter_by(
            doctor_id=current_user.id, patient_id=p.id
        ).filter(ConnectionRequest.status.in_(['pending', 'accepted'])).first()

        result.append({
            'user': p.to_dict(),
            'connection_status': existing.status if existing else 'none',
            'is_connected': p.assigned_doctor_id == current_user.id
        })
    return jsonify({'patients': result})


@doctor_bp.route('/api/doctor/connect/<int:patient_id>', methods=['POST'])
@login_required
@doctor_required
def send_connection_request(patient_id):
    """Send a connection request to a patient."""
    patient = User.query.get(patient_id)
    if not patient or patient.role != 'patient':
        return jsonify({'error': 'Patient not found'}), 404

    # Check if already connected
    if patient.assigned_doctor_id == current_user.id:
        return jsonify({'error': 'Already connected to this patient'}), 400

    # Check for existing pending request
    existing = ConnectionRequest.query.filter_by(
        doctor_id=current_user.id, patient_id=patient_id, status='pending'
    ).first()
    if existing:
        return jsonify({'error': 'Request already pending'}), 400

    data = request.get_json() or {}
    req = ConnectionRequest(
        doctor_id=current_user.id,
        patient_id=patient_id,
        message=data.get('message', f'Dr. {current_user.username} would like to connect with you for therapy support.')
    )
    db.session.add(req)
    db.session.commit()

    # Push real-time SSE event to patient
    try:
        current_app.push_event(patient_id, 'connection_request', req.to_dict())
    except:
        pass

    return jsonify({'message': 'Connection request sent', 'request': req.to_dict()}), 201


# ========== Patient-side connection endpoints ==========

@doctor_bp.route('/api/patient/connection-requests', methods=['GET'])
@login_required
def get_patient_connection_requests():
    """Get pending connection requests for a patient."""
    requests = ConnectionRequest.query.filter_by(
        patient_id=current_user.id, status='pending'
    ).order_by(ConnectionRequest.created_at.desc()).all()
    return jsonify({'requests': [r.to_dict() for r in requests]})


@doctor_bp.route('/api/patient/connection-requests/<int:request_id>/respond', methods=['POST'])
@login_required
def respond_to_connection(request_id):
    """Accept or reject a connection request."""
    conn_req = ConnectionRequest.query.get(request_id)
    if not conn_req or conn_req.patient_id != current_user.id:
        return jsonify({'error': 'Request not found'}), 404

    data = request.get_json()
    action = data.get('action', '')  # 'accept' or 'reject'

    if action == 'accept':
        conn_req.status = 'accepted'
        conn_req.responded_at = datetime.utcnow()
        # Assign doctor to patient
        current_user.assigned_doctor_id = conn_req.doctor_id
        db.session.commit()

        # Push SSE event to doctor (patient connected)
        try:
            current_app.push_event(conn_req.doctor_id, 'connection_accepted', {
                'patient_id': current_user.id,
                'patient_name': current_user.username
            })
        except:
            pass

        return jsonify({
            'message': f'Connected with Dr. {conn_req.doctor.username}',
            'doctor_name': conn_req.doctor.username
        })
    elif action == 'reject':
        conn_req.status = 'rejected'
        conn_req.responded_at = datetime.utcnow()
        db.session.commit()
        return jsonify({'message': 'Request rejected'})
    else:
        return jsonify({'error': 'Invalid action. Use accept or reject.'}), 400


# ========== Doctor Dashboard ==========

@doctor_bp.route('/api/doctor/patients', methods=['GET'])
@login_required
@doctor_required
def get_patients():
    patients = User.query.filter_by(assigned_doctor_id=current_user.id, role='patient').all()
    patient_list = []
    for p in patients:
        latest_report = Report.query.filter_by(user_id=p.id).order_by(Report.created_at.desc()).first()
        active_convs = Conversation.query.filter_by(user_id=p.id, is_active=True).count()
        total_convs = Conversation.query.filter_by(user_id=p.id).count()
        unread_alerts = Alert.query.filter_by(user_id=p.id, doctor_id=current_user.id, is_read=False).count()

        # Get all reports for trend data
        all_reports = Report.query.filter_by(user_id=p.id).order_by(Report.created_at.asc()).all()

        patient_list.append({
            'user': p.to_dict(),
            'latest_report': latest_report.to_dict() if latest_report else None,
            'active_conversations': active_convs,
            'total_conversations': total_convs,
            'unread_alerts': unread_alerts,
            'report_history': [r.to_dict() for r in all_reports]
        })

    return jsonify({'patients': patient_list})


@doctor_bp.route('/api/doctor/patient/<int:patient_id>/reports', methods=['GET'])
@login_required
@doctor_required
def get_patient_reports(patient_id):
    patient = User.query.get(patient_id)
    if not patient or patient.assigned_doctor_id != current_user.id:
        return jsonify({'error': 'Patient not found'}), 404

    reports = Report.query.filter_by(user_id=patient_id).order_by(Report.created_at.desc()).all()
    return jsonify({
        'patient': patient.to_dict(),
        'reports': [r.to_dict() for r in reports]
    })


@doctor_bp.route('/api/doctor/patient/<int:patient_id>/conversations', methods=['GET'])
@login_required
@doctor_required
def get_patient_conversations(patient_id):
    patient = User.query.get(patient_id)
    if not patient or patient.assigned_doctor_id != current_user.id:
        return jsonify({'error': 'Patient not found'}), 404

    convs = Conversation.query.filter_by(user_id=patient_id).order_by(Conversation.started_at.desc()).all()
    return jsonify({
        'patient': patient.to_dict(),
        'conversations': [c.to_dict() for c in convs]
    })


@doctor_bp.route('/api/doctor/patient/<int:patient_id>/conversation/<int:conv_id>/messages', methods=['GET'])
@login_required
@doctor_required
def get_patient_conversation_messages(patient_id, conv_id):
    """Let doctor view full conversation messages of a connected patient."""
    patient = User.query.get(patient_id)
    if not patient or patient.assigned_doctor_id != current_user.id:
        return jsonify({'error': 'Patient not found'}), 404

    conv = Conversation.query.get(conv_id)
    if not conv or conv.user_id != patient_id:
        return jsonify({'error': 'Conversation not found'}), 404

    messages = [m.to_dict() for m in conv.messages]
    return jsonify({
        'conversation': conv.to_dict(),
        'messages': messages
    })


# ========== Analytics API ==========

@doctor_bp.route('/api/doctor/patient/<int:patient_id>/analytics', methods=['GET'])
@login_required
@doctor_required
def get_patient_analytics(patient_id):
    """Get comprehensive analytics for a patient (graphs data)."""
    patient = User.query.get(patient_id)
    if not patient or patient.assigned_doctor_id != current_user.id:
        return jsonify({'error': 'Patient not found'}), 404

    # All reports (sentiment over time)
    reports = Report.query.filter_by(user_id=patient_id).order_by(Report.created_at.asc()).all()

    # All messages with sentiment scores
    conversations = Conversation.query.filter_by(user_id=patient_id).all()
    all_messages = []
    crisis_count = 0
    total_messages = 0
    sentiment_sum = 0

    risk_distribution = {'low': 0, 'moderate': 0, 'high': 0, 'critical': 0}

    for conv in conversations:
        for msg in conv.messages:
            if msg.role == 'user':
                total_messages += 1
                sentiment_sum += msg.sentiment_score
                all_messages.append({
                    'timestamp': msg.timestamp.isoformat(),
                    'sentiment_score': msg.sentiment_score,
                    'content_preview': msg.content[:50]
                })

        # Count risk levels
        risk_distribution[conv.risk_level] = risk_distribution.get(conv.risk_level, 0) + 1

    # Count crisis alerts
    crisis_count = Alert.query.filter_by(user_id=patient_id, alert_type='crisis').count()

    avg_sentiment = sentiment_sum / total_messages if total_messages > 0 else 0

    # Sentiment timeline from reports
    sentiment_timeline = [{
        'date': r.created_at.isoformat(),
        'score': r.sentiment_score,
        'risk_level': r.risk_level,
        'trend': r.emotional_trend
    } for r in reports]

    # Message-level sentiment timeline (for finer granularity)
    message_sentiments = [{
        'timestamp': m['timestamp'],
        'score': m['sentiment_score']
    } for m in all_messages[-50:]]  # Last 50 messages

    return jsonify({
        'patient': patient.to_dict(),
        'summary': {
            'total_sessions': len(conversations),
            'total_messages': total_messages,
            'average_sentiment': round(avg_sentiment, 3),
            'crisis_alerts': crisis_count,
            'current_risk': reports[-1].risk_level if reports else 'low',
            'current_trend': reports[-1].emotional_trend if reports else 'stable'
        },
        'risk_distribution': risk_distribution,
        'sentiment_timeline': sentiment_timeline,
        'message_sentiments': message_sentiments
    })


# ========== Alerts ==========

@doctor_bp.route('/api/doctor/alerts', methods=['GET'])
@login_required
@doctor_required
def get_alerts():
    alerts = Alert.query.filter_by(doctor_id=current_user.id).order_by(Alert.created_at.desc()).all()
    return jsonify({'alerts': [a.to_dict() for a in alerts]})


@doctor_bp.route('/api/doctor/alerts/unread', methods=['GET'])
@login_required
@doctor_required
def get_unread_alerts():
    alerts = Alert.query.filter_by(doctor_id=current_user.id, is_read=False).order_by(
        Alert.created_at.desc()).all()
    return jsonify({'alerts': [a.to_dict() for a in alerts], 'count': len(alerts)})


@doctor_bp.route('/api/doctor/alerts/<int:alert_id>/read', methods=['POST'])
@login_required
@doctor_required
def mark_alert_read(alert_id):
    alert = Alert.query.get(alert_id)
    if not alert or alert.doctor_id != current_user.id:
        return jsonify({'error': 'Alert not found'}), 404
    alert.is_read = True
    db.session.commit()
    return jsonify({'message': 'Alert marked as read'})
